homeassistant/setup.py

```diff
@@ -22,9 +22,11 @@
 DATA_DEPS_REQS = "deps_reqs_processed"
 
 SLOW_SETUP_WARNING = 10
-# Since a pip install can run, we wait
-# 30 minutes to timeout
-SLOW_SETUP_MAX_WAIT = 1800
+
+# Since its possible for databases to be
+# upwards of 36GiB (or larger) in the wild
+# we wait up to 3 hours for startup
+SLOW_SETUP_MAX_WAIT = 10800
```

Comment:
This is ok for now, but the fact that we're extending it this long means we need a better solution :)

For the future, we should rewrite this check to:
 - be able to see if a pip install is happening
 - be able to see if a migration is happening

Reply:
It is a compromise for sure, hopefully very few people have databases ever approaching the size where it matters.  4.6GiB seemed like a large database before today 🙈    

Do we have any data on database sizes?  I could start a forum poll if there isn't something better, we would probably get a good enough sample size to know where to focus effort.

Reply:
Marking this off my TODO list since the new timeout system handles this case 👍 👍 👍